### PR TITLE
adding user function added + getting data by login feature added

### PR DESCRIPTION
**Pull request by OmnixTry**
In this feature:
base user , teacher, and student classes were added.
method to add user to database using login, password andaccount type
mechod to get information of the user specified by certain login

### CODE REVIEW

File: source/source/user.cpp [hunk 1]
```diff
@@ -0,0 +1,72 @@
+#include <string>
+#include <iostream>
+#include <fstream>
+#include "user.h"
+using namespace std;
+
+user::user(string logn, string passwd) {
+	login = logn;
+	password = passwd;
+}
```
Comment (Burnyaxa) [inline source/source/user.cpp]: Please, use only 1 type of constructor

File: source/source/user.h [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+// returns false when something went wrong!!!
+// instruction to use function:
+// pass this function arguments which are login, password and type of account
+// types of accounts are teacher and student
+// if function returns false, account type passed to the function is wrong
+
+bool user_data(string logn, string &passwd, string &accounttype, float &lvl);
+
+
+class user
```
Comment (Burnyaxa) [inline source/source/user.h]: Too many empty fields.

File: source/source/user.h [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+};
+
+class teacher : public user
+{
+public:
+	teacher(string logn, string passwd) : user(logn, passwd) {};
+	~teacher() {};
+	friend bool add_user(string logn, string passwd, string accounttype);
+private:
+
```
Comment (Burnyaxa) [inline source/source/user.h]: Unnecessary field.

File: source/source/user.cpp [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+bool add_user(string logn, string passwd, string accounttype) {
+	user *new_user;
+	if (accounttype == "teacher")
+	{
```
Comment (Burnyaxa) [inline source/source/user.cpp]: Everywhere use "{" and "}" on the same level.

File: source/source/user.cpp [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+	string buff;
+	int tmp = 0;
+	while (!user_data_file.eof()) {
+		getline(user_data_file, buff);
+		clogin = buff.substr(buff.find_first_of(' ') + 1, buff.find(' ', buff.find_first_of(' ') + 1) - buff.find_first_of(' ') - 1);
+	
+		if (logn == clogin) {
+			tmp = buff.find_first_of(' ');
+			accounttype = buff.substr(0, tmp);
+			tmp = accounttype.size() + 1 + logn.size() + 1;
```
Comment (Burnyaxa) [inline source/source/user.cpp]: If there are 2 or more words in a variable name, the second and subsequent words must begin with a capital letter. 
```suggestion
			tmp = accountType.size() + 1 + logn.size() + 1;
```

File: source/source/user.h [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+class user
+{
+public:
+	user(string logn, string passwd);
+	~user() {};
+	friend bool add_user(string logn, string passwd, string accounttype);
+protected:
+	virtual void write_to_file();
```
Comment (Siusarna) [inline source/source/user.h]: If you add argument `accountType` for this function, you can will delete method `write_to_file` in  class `student`.

File: source/source/user.cpp [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+	if (accounttype == "teacher")
+	{
+		new_user = new teacher(logn, passwd);
+		new_user->write_to_file();
+		return true;
+	}
+	else if (accounttype == "student")
+	{
+		new_user = new student(logn, passwd);
+		((teacher *)new_user)->write_to_file();
```
Comment (Siusarna) [inline source/source/user.cpp]: See comment for user.h 22line, then call `write_to_file` as `new_user->write_to_file("teacher")`

File: source/source/user.cpp [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+bool user_data(string logn, string &passwd, string &accounttype, float &lvl) {
+	const string dir_file = "..\\data\\users.txt";
+	ifstream user_data_file(dir_file);
+
+	string clogin;
+	string buff;
+	int tmp = 0;
+	while (!user_data_file.eof()) {
```
Comment (Siusarna) [inline source/source/user.cpp]: It will be better if you write separate function, where you processing the buff

File: source/source/user.h [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+using namespace std;
+
+bool add_user(string logn, string passwd, string accounttype);
+// returns false when something went wrong!!!
+// instruction to use function:
+// pass this function arguments which are login, password and type of account
+// types of accounts are teacher and student
+// if function returns false, account type passed to the function is wrong
+
+bool user_data(string logn, string &passwd, string &accounttype, float &lvl);
```
Comment (Siusarna) [inline source/source/user.h]: Sorry, but I can't understand what this function do. Please write instruction or just lets start discuss here

File: source/source/user.h (hunk 6 above)
Comment (Siusarna) [inline source/source/user.h]: If you add argument `string accountType` for this function, you can will delete method `write_to_file` in class `student`.

File: source/source/user.cpp [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+		getline(user_data_file, buff);
+		clogin = buff.substr(buff.find_first_of(' ') + 1, buff.find(' ', buff.find_first_of(' ') + 1) - buff.find_first_of(' ') - 1);
```
Comment (Siusarna) [inline source/source/user.cpp]: It will be better if you write separate function, where you processing the `buff`

File: source/source/user.cpp [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+	else if (accounttype == "student")
+	{
+		new_user = new student(logn, passwd);
+		((teacher *)new_user)->write_to_file();
+		return true;
+	}
+	return false;
+}
+
+void user::write_to_file() {
```
Comment (Siusarna) [inline source/source/user.cpp]: See comment for user.h 22line, then call `write_to_file` as `new_user->write_to_file("teacher")`

File: source/source/user.cpp (hunk 1 above)
Comment (Siusarna) [reply source/source/user.cpp]: > Please, use only 1 type of constructor

What do you want to say that @OmnixTry  use some other constructor or I just don't understand you

File: source/source/user.h [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+
+
+class user
+{
+public:
```
Comment (Siusarna) [inline source/source/user.h]: and methods `getLogin` and `getPassword`

File: source/source/user.h (hunk 9 above)
Comment (OmnixTry) [reply source/source/user.h]: Sorry i forgot to add explanation to commit, I've allready done it.  You'll see the instructon in next commit.

File: source/source/user.cpp [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+bool user_data(string logn, string &passwd, string &accountType, float &lvl) 
+{
+	const string dir_file = "..\\data\\users.txt";
+	ifstream user_data_file(dir_file);
+	
+	string clogin;
+	string buff;
+	int tmp = 0;
```
Comment (Siusarna) [inline source/source/user.cpp]: `tmp` is assigned a value but never used

**Pull request merged**